core/arch/arm/plat-k3/drivers/mailbox.c

```diff
@@ -0,0 +1,161 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Texas Instruments K3 Mailbox Driver
+ * Copyright (C) 2025 Texas Instruments Incorporated - https://www.ti.com/
+ */
+
+#include <io.h>
+#include <kernel/delay.h>
+#include <mm/core_memprot.h>
+#include <mm/core_mmu.h>
+#include <platform_config.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "ti_sci_protocol.h"
+#include "ti_sci_transport.h"
+
+struct ti_k3_mailbox {
```

Comment:
maybe add a _cfg suffix to it, `ti_k3_mailbox_cfg` type to call out that it's storing configuration variables imo.